### PR TITLE
refactor(list): use list hook

### PR DESCRIPTION
**Pull request by eoaksnes**
## What does this pull request change?

* Use the `useList` hook inside `ListPlugin` to handle list operations

### CODE REVIEW

File: packages/dm-core/src/hooks/useList.tsx [hunk 1]
```diff
@@ -52,115 +67,152 @@ export function useList<T>(idReference: string): IUseListReturnType<T> {
...
+          if (attribute.contained) {
+            if (!Array.isArray(response.data)) {
+              throw new Error(
+                `Not an array! Got document ${JSON.stringify(response.data)} `
+              )
+            }
+
             const items = Object.values(response.data).map((data, index) => ({
-              key: crypto.randomUUID(),
+              key: uuidv4(),
```
Comment (soofstad) [inline packages/dm-core/src/hooks/useList.tsx:96]: Why use extra package for uuid when it's supported in stdlib?

**Comment by soofstad** [comment]
Would like to test before merging

File: packages/dm-core/src/hooks/useList.tsx (hunk 1 above)
Comment (eoaksnes) [reply packages/dm-core/src/hooks/useList.tsx]: Done 👍 Let's just use the default crypto instead.

**Comment by henrikbossart** [comment]
Not working for me, I get no data when trying to load the examples.

i'm also getting the most descriptive error message in the world:
Screenshot 2023-10-31 at 10 26 44

**Comment by soofstad** [comment]
Yepp, just tested the same:
Screenshot_20231031_103247

**Review by soofstad — APPROVED** [review]
Tested okey.
Seems to work perfectly :+1:

**Review by henrikbossart — CHANGES_REQUESTED** [review]
This might not be related to this PR, but flagging it as 'request changes' to get attention on it:

If you move the last item one up, then save, then delete the last item, the reference is goneit seems.. 
I'll upload a screencast

**Comment by henrikbossart** [comment]
https://github.com/equinor/dm-core-packages/assets/3105885/32d37473-e95d-4344-9d2c-d0a5f60afffc

**Review by henrikbossart — APPROVED** [review]
👍

**Pull request merged**